echo-telemetry/src/test/groovy/com/netflix/spinnaker/echo/telemetry/TelemetryEventListenerSpec.groovy

```diff
@@ -0,0 +1,112 @@
+package com.netflix.spinnaker.echo.telemetry
+
+
+import com.google.protobuf.util.JsonFormat
+import com.netflix.spinnaker.echo.config.TelemetryConfig
+import com.netflix.spinnaker.echo.model.Event
+import com.netflix.spinnaker.kork.proto.stats.Application
+import com.netflix.spinnaker.kork.proto.stats.Event as EventProto
+import com.netflix.spinnaker.kork.proto.stats.Execution
+import com.netflix.spinnaker.kork.proto.stats.SpinnakerInstance
+import com.netflix.spinnaker.kork.proto.stats.Stage
+import com.netflix.spinnaker.kork.proto.stats.Status
+import retrofit.client.Response
+import spock.lang.Specification
+import spock.lang.Subject
+
+class TelemetryEventListenerSpec extends Specification {
+
+  def service = Mock(TelemetryService)
+
+  def instanceId = "test-instance"
+  def spinnakerVersion = "1.2.3"
+
+  def applicationName = "someApp"
+  def applicationHash = "e40464bf6d04933c6011c29974eb328777669813a76583e5d547941427df686f"
+
+  def executionId = "execution_id"
+  def executionHash = "6d6de5b8d67c11fff6d817ea3e1190bc63857de0329d253b21aef6e5c6bbebf9"
+
+
+  def "send a telemetry event"() {
+    given:
+    def configProps = new TelemetryConfig.TelemetryConfigProps()
+      .setInstanceId(instanceId)
+      .setSpinnakerVersion(spinnakerVersion)
+
+    @Subject
+    def listener = new TelemetryEventListener(service, configProps)
+
+    Event event = new Event(
+      details: [
+        type       : "orca:pipeline:complete",
+        application: applicationName,
+      ],
+      content: [
+        execution: [
+          id     : executionId,
+          type   : "PIPELINE",
+          status : "SUCCEEDED",
+          trigger: [
+            type: "GIT"
+          ],
+          stages : [
+            [
+              type  : "deploy",
+              status: "SUCCEEDED"
+            ],
+            [
+              type  : "wait",
+              status: "TERMINAL"
+            ],
+          ]
+        ]
+      ]
+    )
+
+    when:
+    listener.processEvent(event)
```

Comment:
I'd run this through with an empty details and content, too... that will catch the NPEs hopefully

Reply:
Added event validation test